// pcln-design-system/dist/cjs/Avatar.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Flex = _interopRequireDefault(require("./Flex"));

var _Heading = _interopRequireDefault(require("./Heading"));

var _Text = _interopRequireDefault(require("./Text"));

var _pclnIcons = require("pcln-icons");

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  justify-content: center;\n  align-items: center;\n  border-radius: 50%;\n  width: ", "px;\n  height: ", "px;\n  background-size: ", "px;\n  background-image: url(", ");\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var StyledImage = (0, _styledComponents["default"])(_Flex["default"])(_templateObject(), function (props) {
  return props.size || 40;
}, function (props) {
  return props.size || 40;
}, function (props) {
  return props.size || 40;
}, function (props) {
  return props.src;
});

function Avatar(_ref) {
  var className = _ref.className,
      title = _ref.title,
      subtitle = _ref.subtitle,
      src = _ref.src,
      initials = _ref.initials,
      size = _ref.size,
      color = _ref.color;
  return _react["default"].createElement(_Flex["default"], {
    className: className
  }, _react["default"].createElement(StyledImage, {
    src: src,
    size: size,
    color: color,
    p: 2
  }, !src && initials && _react["default"].createElement(_Text["default"], {
    fontSize: 1
  }, initials.toUpperCase()), !src && !initials && _react["default"].createElement(_pclnIcons.User, null)), title && _react["default"].createElement(_Flex["default"], {
    flexDirection: "column",
    justifyContent: "center",
    ml: 3
  }, _react["default"].createElement(_Heading["default"].h5, {
    m: 0
  }, title), _react["default"].createElement(_Text["default"], {
    color: "text.light",
    fontSize: 0
  }, subtitle)));
}

Avatar.displayName = 'Avatar';
Avatar.propTypes = {
  className: _propTypes["default"].string,
  title: _propTypes["default"].string,
  subtitle: _propTypes["default"].string,
  src: _propTypes["default"].string,
  initials: _propTypes["default"].string,
  size: _propTypes["default"].number,
  color: _propTypes["default"].string
};
Avatar.defaultProps = {
  className: '',
  color: 'primary'
};
var _default = Avatar;
exports["default"] = _default;
