// pcln-design-system/dist/cjs/BreadcrumbLink.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Flex = _interopRequireDefault(require("./Flex"));

var _Link = _interopRequireDefault(require("./Link"));

var _Text = _interopRequireDefault(require("./Text"));

var _utils = require("./utils");

function BreadcrumbLink(_ref) {
  var className = _ref.className,
      isLastChild = _ref.isLastChild,
      href = _ref.href,
      icon = _ref.icon,
      label = _ref.label,
      dsRef = _ref.dsRef,
      onClick = _ref.onClick;
  var linkColor = isLastChild ? 'text.dark' : 'text.light';
  return _react["default"].createElement(_Flex["default"], {
    className: className,
    alignItems: "center"
  }, icon, _react["default"].createElement(_Link["default"], {
    href: href,
    color: linkColor,
    onClick: onClick,
    dsRef: dsRef
  }, label), !isLastChild && _react["default"].createElement(_Text["default"], {
    m: 2,
    color: "text.light"
  }, "/"));
}

BreadcrumbLink.displayName = 'BreadcrumbLink';
BreadcrumbLink.propTypes = {
  className: _propTypes["default"].string,
  isLastChild: _propTypes["default"].bool,
  href: _propTypes["default"].string,
  icon: _propTypes["default"].node,
  label: _propTypes["default"].string,
  dsRef: _utils.refPropType,
  onClick: _propTypes["default"].func
};
BreadcrumbLink.defaultProps = {
  className: ''
};
var _default = BreadcrumbLink;
exports["default"] = _default;
