// pcln-design-system/dist/cjs/Card.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Box = _interopRequireDefault(require("./Box"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledSystem = require("styled-system");

var _utils = require("./utils");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ", " ", " ", "\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var boxShadow = function boxShadow(props) {
  var boxShadows = {
    sm: {
      'box-shadow': props.theme.boxShadows[0]
    },
    md: {
      'box-shadow': props.theme.boxShadows[1]
    },
    lg: {
      'box-shadow': props.theme.boxShadows[2]
    },
    xl: {
      'box-shadow': props.theme.boxShadows[3]
    }
  };
  return boxShadows[props.boxShadowSize];
};

var boxBorder = function boxBorder(_ref) {
  var borderWidth = _ref.borderWidth,
      color = _ref.color,
      borderColor = _ref.borderColor,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["borderWidth", "color", "borderColor"]);
  return {
    border: borderWidth === 0 ? '0' : "".concat(borderWidth, "px solid ").concat((0, _utils.getPaletteColor)(borderColor, 'base')(props))
  };
};

var Card = (0, _styledComponents["default"])(_Box["default"])(_templateObject(), boxShadow, boxBorder, _styledSystem.borderRadius, (0, _utils.applyVariations)('Card'));
Card.propTypes = _objectSpread({}, _styledSystem.borderRadius.propTypes, {
  boxShadowSize: _propTypes["default"].oneOf(['sm', 'md', 'lg', 'xl']),
  borderColor: (0, _utils.deprecatedColorValue)(),
  color: (0, _utils.deprecatedColorValue)(),
  borderWidth: _propTypes["default"].oneOf([0, 1, 2])
});
Card.defaultProps = {
  borderColor: 'border',
  borderRadius: 1,
  borderWidth: 1
};
Card.displayName = 'Card';
var _default = Card;
exports["default"] = _default;
