// pcln-design-system/dist/cjs/CloseButton.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _IconButton = _interopRequireDefault(require("./IconButton"));

var _pclnIcons = require("pcln-icons");

var _utils = require("./utils");

var CloseButton = function CloseButton(props) {
  return _react["default"].createElement(_IconButton["default"], (0, _extends2["default"])({}, props, {
    icon: _react["default"].createElement(_pclnIcons.Close, {
      size: props.size
    })
  }));
};

CloseButton.defaultProps = {
  size: 24,
  title: 'close'
};
CloseButton.propTypes = {
  onClick: _propTypes["default"].func,
  size: _propTypes["default"].number,
  dsRef: _utils.refPropType,
  title: _propTypes["default"].string
};
CloseButton.displayName = 'CloseButton';
var _default = CloseButton;
exports["default"] = _default;
