// pcln-design-system/dist/cjs/Flag.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _styledSystem = require("styled-system");

var _Flex = _interopRequireDefault(require("./Flex"));

var _Hide = _interopRequireDefault(require("./Hide"));

var _Box = _interopRequireDefault(require("./Box"));

var _utils = require("./utils");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ", "\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  position: relative;\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  font-size: ", "px;\n  border-radius: 0 0 ", " 0;\n  ", " ", ";\n  z-index: 2;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  flex: none;\n  background-color: ", ";\n  border-radius: 0 ", " ", " 0;\n  /* for 32 x 8 triangle */\n  transform: skew(-14deg);\n  position: relative;\n  z-index: 1;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  width: 8px;\n  height: 8px;\n  align-self: flex-end;\n  background-color: inherit;\n  ", ";\n  position: absolute;\n  bottom: 0;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var capitalize = function capitalize(str) {
  return str.charAt(0).toUpperCase() + str.slice(1);
};

var shadowColor = function shadowColor(props) {
  var darkColor = (0, _utils.hasPaletteColor)(props) ? (0, _utils.getPaletteColor)('dark')(props) : (0, _styledSystem.themeGet)("colors.dark".concat(capitalize(props.color)))(props);
  return {
    backgroundImage: !darkColor ? "\n        linear-gradient(45deg, transparent 50%, rgba(0, 0, 0, 0.5) 50%),\n        linear-gradient(45deg, transparent 50%, ".concat(props.color, " 50%)\n      ") : "linear-gradient(45deg, transparent 50%, ".concat(darkColor, " 50%)")
  };
};

var FlagShadow = (0, _styledComponents["default"])(_Box["default"])(_templateObject(), shadowColor);
var FlagRight = (0, _styledComponents["default"])(_Box["default"])(_templateObject2(), function (props) {
  return (0, _utils.getPaletteColor)('base')(props) || props.color;
}, (0, _styledSystem.themeGet)('radius'), (0, _styledSystem.themeGet)('radius'));

var flexAuto = function flexAuto(props) {
  return props.flexAuto ? {
    flex: '1 1 auto'
  } : null;
};

var FlagBody = (0, _styledComponents["default"])(_Box["default"])(_templateObject3(), (0, _styledSystem.themeGet)('fontSizes.0'), (0, _styledSystem.themeGet)('radius'), flexAuto, _utils.color);
var RelativeHide = (0, _styledComponents["default"])(_Hide["default"])(_templateObject4());
var StyledFlex = (0, _styledComponents["default"])(_Flex["default"])(_templateObject5(), (0, _utils.applyVariations)('Flag'));

var Flag = function Flag(_ref) {
  var color = _ref.color,
      bg = _ref.bg,
      children = _ref.children,
      width = _ref.width,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["color", "bg", "children", "width"]);
  return _react["default"].createElement(StyledFlex, (0, _extends2["default"])({
    width: width
  }, props, {
    ml: [0, -2]
  }), _react["default"].createElement(RelativeHide, {
    xs: true
  }, _react["default"].createElement(FlagShadow, {
    width: "4px",
    mr: -2,
    mb: -2,
    color: (0, _utils.hasPaletteColor)(_objectSpread({
      color: color
    }, props)) ? color : bg
  })), _react["default"].createElement(FlagBody, {
    flexAuto: !!width,
    color: color,
    bg: (0, _utils.hasPaletteColor)(_objectSpread({
      color: color
    }, props)) ? false : bg,
    py: [1, 2],
    pl: [1, 3]
  }, children), _react["default"].createElement(FlagRight, {
    width: "18px",
    color: (0, _utils.hasPaletteColor)(_objectSpread({
      color: color
    }, props)) ? color : bg,
    ml: -2
  }));
};

Flag.propTypes = {
  color: (0, _utils.deprecatedColorValue)(),
  bg: (0, _utils.deprecatedPropType)('color')
};
Flag.defaultProps = {
  color: 'white',
  bg: 'green'
};
Flag.displayName = 'Flag';

var _default = (0, _styledComponents.withTheme)(Flag);

exports["default"] = _default;
