// pcln-design-system/dist/cjs/Hug.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _Box = _interopRequireDefault(require("./Box"));

var _Card = _interopRequireDefault(require("./Card"));

var _Flex = _interopRequireDefault(require("./Flex"));

var _Text = _interopRequireDefault(require("./Text"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _utils = require("./utils");

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  & > * {\n    border: 0 !important;\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  border-top-left-radius: ", ";\n  border-top-right-radius: ", ";\n  border-bottom-right-radius: 0;\n  border-bottom-left-radius: 0;\n  overflow: hidden;\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var HugCard = (0, _styledComponents["default"])(_Card["default"])(_templateObject(), function (props) {
  return props.theme.radius;
}, function (props) {
  return props.theme.radius;
}, (0, _utils.applyVariations)('Hug'));
var BorderConcealer = (0, _styledComponents["default"])(_Box["default"])(_templateObject2());

var Hug = function Hug(_ref) {
  var bg = _ref.bg,
      color = _ref.color,
      p = _ref.p,
      fontSize = _ref.fontSize,
      icon = _ref.icon,
      iconDisplay = _ref.iconDisplay,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["bg", "color", "p", "fontSize", "icon", "iconDisplay"]);
  var iconClone;

  if (_react["default"].isValidElement(icon)) {
    iconClone = _react["default"].cloneElement(icon, {
      style: {
        display: iconDisplay,
        mr: 2,
        mt: '-2px',
        mb: '2px',
        size: 24
      }
    });
  }

  return _react["default"].createElement(HugCard, (0, _extends2["default"])({}, props, {
    borderColor: bg || color
  }), _react["default"].createElement(_Flex["default"], {
    bg: bg,
    color: color,
    p: p,
    alignItems: "center"
  }, !!iconClone && iconClone, _react["default"].createElement(_Text["default"], {
    fontSize: fontSize
  }, props.text)), _react["default"].createElement(BorderConcealer, null, props.children));
};

Hug.defaultProps = {
  borderWidth: 1,
  color: 'text.lightest',
  fontSize: 1,
  p: 2
};
Hug.propTypes = {
  iconDisplay: _propTypes["default"].arrayOf(_propTypes["default"].string),
  icon: _propTypes["default"].node,
  text: _propTypes["default"].oneOfType([_propTypes["default"].node, _propTypes["default"].arrayOf(_propTypes["default"].node), _propTypes["default"].string]),
  color: (0, _utils.deprecatedColorValue)()
};
var _default = Hug;
exports["default"] = _default;
