// pcln-design-system/dist/cjs/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  Absolute: true,
  Avatar: true,
  BackgroundImage: true,
  Badge: true,
  Banner: true,
  BlockLink: true,
  Box: true,
  Breadcrumbs: true,
  Button: true,
  Card: true,
  CloseButton: true,
  Container: true,
  Divider: true,
  Flag: true,
  Flex: true,
  Heading: true,
  Hide: true,
  Hug: true,
  Icon: true,
  IconButton: true,
  IconField: true,
  Image: true,
  Input: true,
  InputField: true,
  FormField: true,
  InputGroup: true,
  Label: true,
  Link: true,
  PlaceholderImage: true,
  Radio: true,
  RatingBadge: true,
  Relative: true,
  Select: true,
  SrOnly: true,
  Stamp: true,
  Stepper: true,
  Text: true,
  TextArea: true,
  Tooltip: true,
  ToggleBadge: true,
  Truncate: true,
  Checkbox: true,
  theme: true,
  ThemeProvider: true,
  createTheme: true,
  borders: true,
  color: true,
  deprecatedPropType: true,
  getContrastRatio: true,
  getPaletteColor: true,
  getTextColorOn: true,
  hasPaletteColor: true
};
Object.defineProperty(exports, "Absolute", {
  enumerable: true,
  get: function get() {
    return _Absolute["default"];
  }
});
Object.defineProperty(exports, "Avatar", {
  enumerable: true,
  get: function get() {
    return _Avatar["default"];
  }
});
Object.defineProperty(exports, "BackgroundImage", {
  enumerable: true,
  get: function get() {
    return _BackgroundImage["default"];
  }
});
Object.defineProperty(exports, "Badge", {
  enumerable: true,
  get: function get() {
    return _Badge["default"];
  }
});
Object.defineProperty(exports, "Banner", {
  enumerable: true,
  get: function get() {
    return _Banner["default"];
  }
});
Object.defineProperty(exports, "BlockLink", {
  enumerable: true,
  get: function get() {
    return _BlockLink["default"];
  }
});
Object.defineProperty(exports, "Box", {
  enumerable: true,
  get: function get() {
    return _Box["default"];
  }
});
Object.defineProperty(exports, "Breadcrumbs", {
  enumerable: true,
  get: function get() {
    return _Breadcrumbs["default"];
  }
});
Object.defineProperty(exports, "Button", {
  enumerable: true,
  get: function get() {
    return _Button["default"];
  }
});
Object.defineProperty(exports, "Card", {
  enumerable: true,
  get: function get() {
    return _Card["default"];
  }
});
Object.defineProperty(exports, "CloseButton", {
  enumerable: true,
  get: function get() {
    return _CloseButton["default"];
  }
});
Object.defineProperty(exports, "Container", {
  enumerable: true,
  get: function get() {
    return _Container["default"];
  }
});
Object.defineProperty(exports, "Divider", {
  enumerable: true,
  get: function get() {
    return _Divider["default"];
  }
});
Object.defineProperty(exports, "Flag", {
  enumerable: true,
  get: function get() {
    return _Flag["default"];
  }
});
Object.defineProperty(exports, "Flex", {
  enumerable: true,
  get: function get() {
    return _Flex["default"];
  }
});
Object.defineProperty(exports, "Heading", {
  enumerable: true,
  get: function get() {
    return _Heading["default"];
  }
});
Object.defineProperty(exports, "Hide", {
  enumerable: true,
  get: function get() {
    return _Hide["default"];
  }
});
Object.defineProperty(exports, "Hug", {
  enumerable: true,
  get: function get() {
    return _Hug["default"];
  }
});
Object.defineProperty(exports, "Icon", {
  enumerable: true,
  get: function get() {
    return _Icon["default"];
  }
});
Object.defineProperty(exports, "IconButton", {
  enumerable: true,
  get: function get() {
    return _IconButton["default"];
  }
});
Object.defineProperty(exports, "IconField", {
  enumerable: true,
  get: function get() {
    return _IconField["default"];
  }
});
Object.defineProperty(exports, "Image", {
  enumerable: true,
  get: function get() {
    return _Image["default"];
  }
});
Object.defineProperty(exports, "Input", {
  enumerable: true,
  get: function get() {
    return _Input["default"];
  }
});
Object.defineProperty(exports, "InputField", {
  enumerable: true,
  get: function get() {
    return _FormField["default"];
  }
});
Object.defineProperty(exports, "FormField", {
  enumerable: true,
  get: function get() {
    return _FormField["default"];
  }
});
Object.defineProperty(exports, "InputGroup", {
  enumerable: true,
  get: function get() {
    return _InputGroup["default"];
  }
});
Object.defineProperty(exports, "Label", {
  enumerable: true,
  get: function get() {
    return _Label["default"];
  }
});
Object.defineProperty(exports, "Link", {
  enumerable: true,
  get: function get() {
    return _Link["default"];
  }
});
Object.defineProperty(exports, "PlaceholderImage", {
  enumerable: true,
  get: function get() {
    return _PlaceholderImage["default"];
  }
});
Object.defineProperty(exports, "Radio", {
  enumerable: true,
  get: function get() {
    return _Radio["default"];
  }
});
Object.defineProperty(exports, "RatingBadge", {
  enumerable: true,
  get: function get() {
    return _RatingBadge["default"];
  }
});
Object.defineProperty(exports, "Relative", {
  enumerable: true,
  get: function get() {
    return _Relative["default"];
  }
});
Object.defineProperty(exports, "Select", {
  enumerable: true,
  get: function get() {
    return _Select["default"];
  }
});
Object.defineProperty(exports, "SrOnly", {
  enumerable: true,
  get: function get() {
    return _SrOnly["default"];
  }
});
Object.defineProperty(exports, "Stamp", {
  enumerable: true,
  get: function get() {
    return _Stamp["default"];
  }
});
Object.defineProperty(exports, "Stepper", {
  enumerable: true,
  get: function get() {
    return _Stepper["default"];
  }
});
Object.defineProperty(exports, "Text", {
  enumerable: true,
  get: function get() {
    return _Text["default"];
  }
});
Object.defineProperty(exports, "TextArea", {
  enumerable: true,
  get: function get() {
    return _TextArea["default"];
  }
});
Object.defineProperty(exports, "Tooltip", {
  enumerable: true,
  get: function get() {
    return _Tooltip["default"];
  }
});
Object.defineProperty(exports, "ToggleBadge", {
  enumerable: true,
  get: function get() {
    return _ToggleBadge["default"];
  }
});
Object.defineProperty(exports, "Truncate", {
  enumerable: true,
  get: function get() {
    return _Truncate["default"];
  }
});
Object.defineProperty(exports, "Checkbox", {
  enumerable: true,
  get: function get() {
    return _Checkbox["default"];
  }
});
Object.defineProperty(exports, "theme", {
  enumerable: true,
  get: function get() {
    return _theme["default"];
  }
});
Object.defineProperty(exports, "ThemeProvider", {
  enumerable: true,
  get: function get() {
    return _ThemeProvider["default"];
  }
});
Object.defineProperty(exports, "createTheme", {
  enumerable: true,
  get: function get() {
    return _createTheme["default"];
  }
});
Object.defineProperty(exports, "borders", {
  enumerable: true,
  get: function get() {
    return _utils.borders;
  }
});
Object.defineProperty(exports, "color", {
  enumerable: true,
  get: function get() {
    return _utils.color;
  }
});
Object.defineProperty(exports, "deprecatedPropType", {
  enumerable: true,
  get: function get() {
    return _utils.deprecatedPropType;
  }
});
Object.defineProperty(exports, "getContrastRatio", {
  enumerable: true,
  get: function get() {
    return _utils.getContrastRatio;
  }
});
Object.defineProperty(exports, "getPaletteColor", {
  enumerable: true,
  get: function get() {
    return _utils.getPaletteColor;
  }
});
Object.defineProperty(exports, "getTextColorOn", {
  enumerable: true,
  get: function get() {
    return _utils.getTextColorOn;
  }
});
Object.defineProperty(exports, "hasPaletteColor", {
  enumerable: true,
  get: function get() {
    return _utils.hasPaletteColor;
  }
});

var _Absolute = _interopRequireDefault(require("./Absolute"));

var _Avatar = _interopRequireDefault(require("./Avatar"));

var _BackgroundImage = _interopRequireDefault(require("./BackgroundImage"));

var _Badge = _interopRequireDefault(require("./Badge"));

var _Banner = _interopRequireDefault(require("./Banner"));

var _BlockLink = _interopRequireDefault(require("./BlockLink"));

var _Box = _interopRequireDefault(require("./Box"));

var _Breadcrumbs = _interopRequireDefault(require("./Breadcrumbs"));

var _Button = _interopRequireDefault(require("./Button"));

var _Card = _interopRequireDefault(require("./Card"));

var _CloseButton = _interopRequireDefault(require("./CloseButton"));

var _Container = _interopRequireDefault(require("./Container"));

var _Divider = _interopRequireDefault(require("./Divider"));

var _Flag = _interopRequireDefault(require("./Flag"));

var _Flex = _interopRequireDefault(require("./Flex"));

var _Heading = _interopRequireDefault(require("./Heading"));

var _Hide = _interopRequireDefault(require("./Hide"));

var _Hug = _interopRequireDefault(require("./Hug"));

var _Icon = _interopRequireDefault(require("./Icon"));

var _IconButton = _interopRequireDefault(require("./IconButton"));

var _IconField = _interopRequireDefault(require("./IconField"));

var _Image = _interopRequireDefault(require("./Image"));

var _Input = _interopRequireDefault(require("./Input"));

var _FormField = _interopRequireDefault(require("./FormField"));

var _InputGroup = _interopRequireDefault(require("./InputGroup"));

var _Label = _interopRequireDefault(require("./Label"));

var _Link = _interopRequireDefault(require("./Link"));

var _PlaceholderImage = _interopRequireDefault(require("./PlaceholderImage"));

var _Radio = _interopRequireDefault(require("./Radio"));

var _RatingBadge = _interopRequireDefault(require("./RatingBadge"));

var _Relative = _interopRequireDefault(require("./Relative"));

var _Select = _interopRequireDefault(require("./Select"));

var _SrOnly = _interopRequireDefault(require("./SrOnly"));

var _Stamp = _interopRequireDefault(require("./Stamp"));

var _Stepper = _interopRequireDefault(require("./Stepper"));

var _Text = _interopRequireDefault(require("./Text"));

var _TextArea = _interopRequireDefault(require("./TextArea"));

var _Tooltip = _interopRequireDefault(require("./Tooltip"));

var _ToggleBadge = _interopRequireDefault(require("./ToggleBadge"));

var _Truncate = _interopRequireDefault(require("./Truncate"));

var _Checkbox = _interopRequireDefault(require("./Checkbox"));

var _theme = _interopRequireWildcard(require("./theme"));

Object.keys(_theme).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _theme[key];
    }
  });
});

var _ThemeProvider = _interopRequireDefault(require("./ThemeProvider"));

var _createTheme = _interopRequireDefault(require("./createTheme"));

var _utils = require("./utils");
