// pcln-design-system/dist/cjs/Link.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _styledSystem = require("styled-system");

var _utils = require("./utils");

var _Button = require("./Button");

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ", " ", ";\n  ", "\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    &:hover {\n      text-decoration: none;\n    }\n\n    ", "\n  "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    cursor: pointer;\n    text-decoration: none;\n    color: ", ";\n\n    &:hover {\n      color: ", ";\n      text-decoration: underline;\n    }\n  "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    &:hover {\n      text-decoration: none;\n    }\n\n    ", "\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var variations = {
  fill: (0, _styledComponents.css)(_templateObject(), _Button.buttonStyles),
  link: (0, _styledComponents.css)(_templateObject2(), (0, _utils.getPaletteColor)('base'), (0, _utils.getPaletteColor)('dark')),
  outline: (0, _styledComponents.css)(_templateObject3(), _Button.buttonStyles)
};
var Link = (0, _utils.mapProps)(function (_ref) {
  var _objectSpread2;

  var target = _ref.target,
      dsRef = _ref.dsRef,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["target", "dsRef"]);
  return _objectSpread((_objectSpread2 = {}, (0, _defineProperty2["default"])(_objectSpread2, (0, _utils.getSCMigrationRef)(), dsRef), (0, _defineProperty2["default"])(_objectSpread2, "rel", target === '_blank' ? 'noopener' : null), (0, _defineProperty2["default"])(_objectSpread2, "target", target), _objectSpread2), props);
})(_styledComponents["default"].a(_templateObject4(), _styledSystem.width, _styledSystem.space, (0, _utils.applyVariations)('Link', variations)));
Link.displayName = 'Link';
Link.defaultProps = {
  color: 'primary',
  variation: 'link'
};
Link.propTypes = {
  color: (0, _utils.deprecatedColorValue)(),
  dsRef: _utils.refPropType,
  variation: _propTypes["default"].oneOf(Object.keys(variations))
};
var _default = Link;
exports["default"] = _default;
