// pcln-design-system/dist/cjs/PlaceholderImage.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Image = _interopRequireDefault(require("./Image"));

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  height: ", "px;\n  width: ", "px;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var RandomImage = (0, _styledComponents["default"])(_Image["default"])(_templateObject(), function (props) {
  return props.height;
}, function (props) {
  return props.width;
});

var determineSRC = function determineSRC(blur, chooseSrc, height, width) {
  var imageURLs = ["http://placeimg.com/".concat(width, "/").concat(height, "/any"), "http://placekitten.com/".concat(width, "/").concat(height), "https://picsum.photos/".concat(width, "/").concat(height), "https://source.unsplash.com/".concat(width, "x").concat(height), "https://www.fillmurray.com/".concat(width, "/").concat(height)];

  if (blur || chooseSrc) {
    return blur ? "https://picsum.photos/".concat(width, "/").concat(height, "?blur") : imageURLs[chooseSrc];
  }

  var randomNum = Math.floor(Math.random() * imageURLs.length);
  return imageURLs[randomNum];
};

var PlaceholderImage = function PlaceholderImage(props) {
  var alt = props.alt,
      ariaHidden = props.ariaHidden,
      blur = props.blur,
      chooseSrc = props.chooseSrc,
      className = props.className,
      height = props.height,
      width = props.width;
  return _react["default"].createElement(RandomImage, {
    alt: alt || 'Random image used for placeholder content',
    "aria-hidden": ariaHidden,
    blur: blur,
    className: className,
    height: height,
    src: determineSRC(blur, chooseSrc, height, width),
    width: width
  });
};

PlaceholderImage.displayName = 'PlaceholderImage';
PlaceholderImage.propTypes = {
  ariaHidden: _propTypes["default"].bool,
  blur: _propTypes["default"].bool,
  chooseSrc: _propTypes["default"].string,
  height: _propTypes["default"].string,
  width: _propTypes["default"].string
};
PlaceholderImage.defaultProps = {
  ariaHidden: true,
  blur: false,
  height: '500',
  width: '500'
};
var _default = PlaceholderImage;
exports["default"] = _default;
