// pcln-design-system/dist/cjs/Radio.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _pclnIcons = require("pcln-icons");

var _utils = require("./utils");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  vertical-align: middle;\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  vertical-align: middle;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  appearance: none;\n  opacity: 0;\n  position: absolute;\n  z-index: 0;\n\n  & ~ svg {\n    border: 1px solid transparent;\n    border-radius: 50%;\n    padding: 2px;\n  }\n\n  &:focus {\n    box-shadow: none;\n    & ~ svg {\n      border: 1px solid ", ";\n      background-color: ", ";\n    }\n  }\n  &:checked ~ svg {\n    color: ", ";\n  }\n  &:disabled ~ svg {\n    color: ", ";\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: inline-block;\n  color: ", ";\n  &:hover > svg {\n    ", ";\n  }\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var RadioWrap = _styledComponents["default"].div(_templateObject(), (0, _utils.getPaletteColor)('border.base'), function (props) {
  if (props.checked && !props.disabled) {
    return "color: ".concat((0, _utils.getPaletteColor)('dark')(props), " !important;");
  }

  return props.disabled ? null : "color: ".concat((0, _utils.getPaletteColor)('base')(props), ";");
}, (0, _utils.applyVariations)('Radio'));

var RadioInput = _styledComponents["default"].input(_templateObject2(), (0, _utils.getPaletteColor)('border.base'), (0, _utils.getPaletteColor)('background.light'), (0, _utils.getPaletteColor)('base'), (0, _utils.getPaletteColor)('border.base'));

var RadioCheckedIcon = (0, _styledComponents["default"])(_pclnIcons.RadioChecked)(_templateObject3());
var RadioEmptyIcon = (0, _styledComponents["default"])(_pclnIcons.RadioEmpty)(_templateObject4());

var RadioIcon = function RadioIcon(_ref) {
  var checked = _ref.checked,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["checked"]);
  return checked ? _react["default"].createElement(RadioCheckedIcon, props) : _react["default"].createElement(RadioEmptyIcon, props);
};

var Radio = function Radio(props) {
  var checked = props.checked,
      disabled = props.disabled,
      size = props.size;
  var borderAdjustedSize = size + 4; // TODO remove once migrated to SC4

  var spreadProps = _objectSpread({}, props);

  spreadProps[(0, _utils.getSCMigrationRef)()] = props.dsRef;
  delete spreadProps.dsRef;
  return _react["default"].createElement(RadioWrap, {
    color: props.color,
    checked: checked,
    disabled: disabled
  }, _react["default"].createElement(RadioInput, (0, _extends2["default"])({
    type: "radio"
  }, spreadProps)), _react["default"].createElement(RadioIcon, {
    checked: checked,
    size: borderAdjustedSize
  }));
};

Radio.defaultProps = {
  color: 'primary',
  size: 24
};
Radio.propTypes = {
  color: (0, _utils.deprecatedColorValue)(),
  dsRef: _utils.refPropType,
  size: _propTypes["default"].number
};
var _default = Radio;
exports["default"] = _default;
