// pcln-design-system/dist/cjs/Select.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _pclnIcons = require("pcln-icons");

var _utils = require("./utils");

var _Flex = _interopRequireDefault(require("./Flex"));

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])([""]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  appearance: none;\n  display: block;\n  width: 100%;\n  font-family: inherit;\n  color: inherit;\n  background-color: transparent;\n  border-radius: ", ";\n  border-width: 1px;\n  border-style: solid;\n\n  ", " ", " ", "\n  ::-ms-expand {\n    display: none;\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  pointer-events: none;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var ClickableIcon = (0, _styledComponents["default"])(_pclnIcons.ChevronDown)(_templateObject());
var SelectBase = (0, _utils.mapProps)(function (_ref) {
  var dsRef = _ref.dsRef,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["dsRef"]);
  return _objectSpread((0, _defineProperty2["default"])({}, (0, _utils.getSCMigrationRef)(), dsRef), props);
})(_styledComponents["default"].select(_templateObject2(), (0, _styledSystem.themeGet)('radius'), _utils.borders, _styledSystem.space, _styledSystem.fontSize));
SelectBase.defaultProps = {
  fontSize: [2, null, 1],
  m: 0,
  pl: 12,
  pr: 32,
  py: 14
};
SelectBase.propTypes = _objectSpread({
  dsRef: _utils.refPropType
}, _styledSystem.space.propTypes, {}, _styledSystem.fontSize.propTypes, {
  color: (0, _utils.deprecatedColorValue)()
}, _utils.borders.propTypes);
var Select = (0, _styledComponents["default"])(function (props) {
  return _react["default"].createElement(_Flex["default"], {
    width: 1,
    alignItems: "center"
  }, _react["default"].createElement(SelectBase, props), _react["default"].createElement(ClickableIcon, {
    ml: -32,
    color: "text.light"
  }));
})(_templateObject3());
Select.isField = true;
var _default = Select;
exports["default"] = _default;
