// pcln-design-system/dist/cjs/Text.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.textShadow = exports.bold = exports.regular = exports.caps = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledSystem = require("styled-system");

var _utils = require("./utils");

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  color: ", ";\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var caps = function caps(props) {
  return props.caps ? {
    textTransform: 'uppercase',
    letterSpacing: (0, _styledSystem.themeGet)('letterSpacings.caps')(props)
  } : null;
};

exports.caps = caps;

var regular = function regular(props) {
  return props.regular ? {
    fontWeight: props.theme.regular
  } : null;
};

exports.regular = regular;

var bold = function bold(props) {
  return props.bold ? {
    fontWeight: props.theme.bold
  } : null;
};

exports.bold = bold;

var textShadow = function textShadow(props) {
  var textShadowSize = props.textShadowSize || 'md';
  return props.enableTextShadow ? {
    textShadow: props.theme.textShadows[textShadowSize]
  } : null;
};

exports.textShadow = textShadow;
var Text = (0, _utils.mapProps)(function (_ref) {
  var align = _ref.align,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["align"]);
  return _objectSpread({
    textAlign: align
  }, props);
})(_styledComponents["default"].div(_templateObject(), (0, _utils.getPaletteColor)('base'), function (props) {
  return props.bg ? "background-color: ".concat((0, _utils.getPaletteColor)(props.bg, 'base')(props), ";") : '';
}, (0, _utils.applyVariations)('Text'), _styledSystem.textStyle, _styledSystem.fontSize, _styledSystem.fontWeight, _styledSystem.textAlign, _styledSystem.lineHeight, _styledSystem.space, caps, regular, bold, textShadow));
Text.displayName = 'Text';
Text.propTypes = _objectSpread({}, _styledSystem.textStyle.propTypes, {}, _styledSystem.fontSize.propTypes, {}, _styledSystem.fontWeight.propTypes, {}, _styledSystem.textAlign.propTypes, {}, _styledSystem.lineHeight.propTypes, {}, _styledSystem.space.propTypes, {
  color: (0, _utils.deprecatedColorValue)(),
  caps: _propTypes["default"].bool,
  regular: _propTypes["default"].bool,
  bold: _propTypes["default"].bool,
  enableTextShadow: _propTypes["default"].bool,
  textShadowSize: _propTypes["default"].oneOf(['sm', 'md']),
  align: (0, _utils.deprecatedPropType)('textAlign')
});
Text.span = Text.withComponent('span');
Text.p = Text.withComponent('p');
Text.s = Text.withComponent('s');
var _default = Text;
exports["default"] = _default;
