// pcln-design-system/dist/cjs/ThemeProvider.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.Base = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _createTheme = _interopRequireDefault(require("./createTheme"));

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  font-family: ", ";\n  line-height: ", ";\n  font-weight: ", ";\n\n  * {\n    box-sizing: border-box;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var Base = _styledComponents["default"].div(_templateObject(), function (props) {
  return props.theme.font;
}, function (props) {
  return props.theme.lineHeights.standard;
}, function (props) {
  return props.theme.fontWeights.medium;
});

exports.Base = Base;

var ThemeProvider = function ThemeProvider(_ref) {
  var theme = _ref.theme,
      customBreakpoints = _ref.customBreakpoints,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["theme", "customBreakpoints"]);
  // Once updated to React 16.8 this should be wrapped in React.useMemo()
  var mergedTheme = (0, _createTheme["default"])(theme, customBreakpoints);
  return _react["default"].createElement(_styledComponents.ThemeProvider, {
    theme: mergedTheme
  }, _react["default"].createElement(Base, props));
};

ThemeProvider.propTypes = {
  /** The theme to drive the look and feel */
  theme: _propTypes["default"].shape({
    breakpoints: _propTypes["default"].arrayOf(_propTypes["default"].string),
    mediaQueries: _propTypes["default"].arrayOf(_propTypes["default"].string),
    space: _propTypes["default"].arrayOf(_propTypes["default"].number),
    font: _propTypes["default"].string,
    fontSizes: _propTypes["default"].arrayOf(_propTypes["default"].number),
    fontWeights: _propTypes["default"].shape({
      medium: _propTypes["default"].number,
      bold: _propTypes["default"].number,
      regular: _propTypes["default"].number
    }),
    lineHeights: _propTypes["default"].shape({
      standard: _propTypes["default"].number,
      display: _propTypes["default"].number
    }),
    letterSpacings: _propTypes["default"].shape({
      normal: _propTypes["default"].string,
      caps: _propTypes["default"].string
    }),
    regular: _propTypes["default"].number,
    bold: _propTypes["default"].number,
    colors: _propTypes["default"].object,
    palette: _propTypes["default"].object,
    radii: _propTypes["default"].arrayOf(_propTypes["default"].number),
    radius: _propTypes["default"].string,
    boxShadows: _propTypes["default"].arrayOf(_propTypes["default"].string),
    maxContainerWidth: _propTypes["default"].string,
    duration: _propTypes["default"].object,
    timingFunctions: _propTypes["default"].object,
    transitionDelays: _propTypes["default"].object
  }),

  /** Array of pixel values for custom breakpoint overrides */
  customBreakpoints: _propTypes["default"].arrayOf(_propTypes["default"].string)
};
var _default = ThemeProvider;
exports["default"] = _default;
