// pcln-design-system/dist/cjs/ToggleBadge.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _utils = require("./utils");

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  border-radius: ", ";\n  border: 0;\n  display: inline-block;\n  font-weight: ", ";\n  font-family: inherit;\n  cursor: pointer;\n  background-color: ", ";\n  color: ", ";\n  ", " ", ";\n  &:hover {\n    background-color: ", ";\n  }\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var ToggleBadge = (0, _utils.mapProps)(function (_ref) {
  var dsRef = _ref.dsRef,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["dsRef"]);
  return _objectSpread((0, _defineProperty2["default"])({}, (0, _utils.getSCMigrationRef)(), dsRef), props);
})(_styledComponents["default"].button(_templateObject(), function (props) {
  return props.theme.radius;
}, function (props) {
  return props.theme.bold;
}, function (props) {
  return props.selected ? (0, _utils.getPaletteColor)(props.bg || props.color, 'light')(props) : props.unSelectedBg;
}, (0, _utils.getPaletteColor)('base'), _styledSystem.space, _styledSystem.fontSize, function (props) {
  return (0, _utils.getPaletteColor)(props.bg || props.color, 'light')(props);
}, (0, _utils.applyVariations)('ToggleBadge')));
ToggleBadge.displayName = 'ToggleBadge';
ToggleBadge.propTypes = _objectSpread({
  selected: _propTypes["default"].bool,
  dsRef: _utils.refPropType
}, _styledSystem.space.propTypes, {}, _styledSystem.fontSize.propTypes, {
  color: (0, _utils.deprecatedColorValue)(),
  bg: (0, _utils.deprecatedPropType)('color')
});
ToggleBadge.defaultProps = {
  selected: false,
  px: 2,
  py: 1,
  mx: 1,
  my: 1,
  fontSize: 0,
  color: 'primary',
  unSelectedBg: 'transparent'
};
var _default = ToggleBadge;
exports["default"] = _default;
